homeassistant/components/smlight/config_flow.py

```diff
@@ -199,7 +202,10 @@ async def _async_complete_entry(
         self, user_input: dict[str, Any]
     ) -> ConfigFlowResult:
         info = await self.client.get_info()
-        await self.async_set_unique_id(format_mac(info.MAC))
+
+        await self.async_set_unique_id(
+            format_mac(info.MAC), raise_on_progress=not self._user_flow
```

Comment:
```suggestion
            format_mac(info.MAC), raise_on_progress=(self.source != SOURCE_USER)
```
should be the more standard way to achieve this